docs/spec/spec.md

```diff
@@ -0,0 +1,1031 @@
+            }
+            lastPoint = p;
+        });
+        decimal endTime = time:monotonicNow();
+        int elapsedTime = <int>(endTime - startTime);
+        return {point_count: pointCount, feature_count: featureCount, distance: distance, elapsed_time: elapsedTime};
+    }
+}
+```
+
+**RPC Using a Caller**
+
+The Ballerina implementation of the client streaming RPC using a caller return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function RecordRoute(RouteGuideRouteSummaryCaller caller, stream<Point, grpc:Error?> clientStream) returns error? {
+        Point? lastPoint = ();
+        int pointCount = 0;
+        int featureCount = 0;
+        int distance = 0;
+
+        decimal startTime = time:monotonicNow();
+        check clientStream.forEach(function(Point p) {
+            pointCount += 1;
+            if pointExistsInFeatures(FEATURES, p) {
+                featureCount += 1;
+            }
+
+            if lastPoint is Point {
+                distance = calculateDistance(<Point>lastPoint, p);
+            }
+            lastPoint = p;
+        });
+        decimal endTime = time:monotonicNow();
+        int elapsedTime = <int>(endTime - startTime);
+        return caller->sendRouteSummary({point_count: pointCount, feature_count: featureCount, distance: distance, elapsed_time: elapsedTime});
+    }
+}
+```
+
+**RPC Invocation**
+
+For each RPC in the protobuf definition, the generated Ballerina stub contains a client. That generated client interacts with the actual RPC service during an RPC call. Unlike the server streaming scenario, the Ballerina client streaming does not use a streaming object to pass data to the client-side because it should allow users to send and receive data asynchronously. Instead, it uses a streaming object to send and receive data from the server.
+```ballerina
+public function main() returns error? {
+    RouteGuideClient ep = check new ("http://localhost:8980");
+    Point[] points = [
+        {latitude: 406109563, longitude: -742186778}, 
+        {latitude: 411733222, longitude: -744228360}, 
+        {latitude: 744228334, longitude: -742186778}
+    ];
+    RecordRouteStreamingClient recordRouteStrmClient = check ep->RecordRoute();
+    foreach Point p in points {
+        check recordRouteStrmClient->sendPoint(p);
+    }
+    check recordRouteStrmClient->complete();
+    RouteSummary? routeSummary = check recordRouteStrmClient->receiveRouteSummary();
+    if routeSummary is RouteSummary {
+        io:println(`Finished trip with ${routeSummary.point_count} points. Passed ${routeSummary.feature_count} features. "Travelled ${routeSummary.distance} meters. It took ${routeSummary.elapsed_time} seconds.`);
+    }
+}
+```
+
+## 4.4. Bidirectional Streaming RPC
+The RPC service definition of a bidirectional streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc ListFeatures(Rectangle) returns (stream Feature) {}
```

Comment:
Seems like the Bidirectional streaming method is incorrect. please check

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/583/commits/ecbfe1acbc43fc9a96e1ed8b9967f3d4e554556f